docs/src/main/tut/datatypes/eithert.md

```diff
@@ -0,0 +1,162 @@
+
+def parseDouble(s: String): Either[String, Double] =
+  Try(s.toDouble).toEither match {
+    case Left(_) => Left(s"$s is not a number")
+    case Right(n) => Right(n)
+  }
+
+def divide(a: Double, b: Double): Either[String, Double] =
+  Either.cond(b != 0, a / b, "Cannot divide by zero")
+
+def divisionProgram(inputA: String, inputB: String): Either[String, Double] =
+  for {
+    a <- parseDouble(inputA)
+    b <- parseDouble(inputB)
+    result <- divide(a, b)
+  } yield result
+
+divisionProgram("4", "2") // Right(2.0)
+divisionProgram("a", "b") // Left("a is not a number")
+```
+
+Suppose `parseDouble` and `divide` are rewritten to be asynchronous and return
+`Future[Either[String, Double]]` instead. The for-comprehension can no longer be
+used since `divisionProgram` must now compose `Future` and `Either` together,
+which means that the error handling must be performed explicitly to ensure that
+the proper types are returned:
+
+```tut:silent
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+def parseDoubleAsync(s: String): Future[Either[String, Double]] =
+  Future.successful(parseDouble(s))
+def divideAsync(a: Double, b: Double): Future[Either[String, Double]] =
+  Future.successful(divide(a, b))
+
+def divisionProgramAsync(inputA: String, inputB: String): Future[Either[String, Double]] =
+  parseDoubleAsync(inputA) flatMap { eitherA =>
+    parseDoubleAsync(inputB) flatMap { eitherB =>
+      val parseResult = for {
+        a <- eitherA
+        b <- eitherB
+      } yield (a, b)
+      
+      parseResult match {
+        case Right((a, b)) => divideAsync(a, b)
+        case l@Left(err) => Future.successful(Left(err))
+      }
+    }
+  }
+```
+
+Clearly, the updated code is less readible and more verbose: the details of the
+program are now mixed with the error handling. In addition, as more `Either`s
+and `Futures` are included, the amount of boilerplate required to properly
+handle the errors will increase dramatically.
+
+## EitherT
+
+`EitherT[F[_], A, B]` is a lightweight wrapper for `F[Either[A, B]]` that makes
+it easy to compose `Either`s and `F`s together. To use `EitherT`, values of
+`Either`, `F`, `A`, and `B` are first converted into `EitherT`, and the
+resulting `EitherT` values are then composed using combinators. For example, the
+asynchronous division program can be rewritten as follows:
+
+```tut:silent
+import cats.data.EitherT
+import cats.implicits._
+
+def divisionProgramAsync(inputA: String, inputB: String): EitherT[Future, String, Double] =
+  for {
+    a <- EitherT(parseDoubleAsync(inputA))
+    b <- EitherT(parseDoubleAsync(inputB))
+    result <- EitherT(divideAsync(a, b))
+  } yield result
+
+divisionProgramAsync("4", "2").value // Future(Right(2.0))
+divisionProgramAsync("a", "b").value // Future(Left("a is not a number"))
+```
+
+Note that since `EitherT` is a monad, monadic combinators such as `flatMap` can
+be used to compose `EitherT` values.
```

Comment:
Maybe clarify here, that `EitherT` forms a Monad, only if `F[_]` is also a Monad :)